library/src/components/__tests__/Schema.test.tsx

```diff
@@ -24,34 +22,8 @@ describe('Schema component', () => {
     };
     schema.properties.circular = schema;
     schema.properties.circularTwo = schema;
-    const schemaModel = new SchemaModel(schema);
-
-    render(<Schema schema={schemaModel} />);
-
-    // properties
-    expect(screen.getByText('nonCircular')).toBeDefined();
-    expect(screen.getByText('circular')).toBeDefined();
-    expect(screen.getByText('circularTwo')).toBeDefined();
-    // [CIRCULAR] annotation
-    expect(screen.getAllByText('object [CIRCULAR]')).toBeDefined();
-    expect(screen.getAllByText('object [CIRCULAR]')).toHaveLength(2);
-  });
-
-  test('should work with circular references in schema - using `x-parser-circular-props` extensions', async () => {
-    const schema = {
-      type: 'object',
-      properties: {
-        nonCircular: {
-          type: 'string',
-        },
-        circular: {},
-        circularTwo: {},
-      },
-    };
-    schema.properties.circular = schema;
-    schema.properties.circularTwo = schema;
-    schema['x-parser-circular-props'] = ['circular', 'circularTwo'];
-    const schemaModel = new SchemaModel(schema);
+    schema['x-schema-private-render-type'] = true;
```

Comment:
I had to add this to render the type for some reason, even though it was not needed in the previous setup. 

I simply have no idea where this is supposed to already have been sat, any idea @magicmatatjahu?

Reply:
I don't remember exactly, but it shouldn't be there, maybe this `if` (https://github.com/asyncapi/asyncapi-react/blob/8cf921d8c1e5b231222d615d242f62144146049e/library/src/components/Schema.tsx#L47) is needed for some extra stuff, I seriously don't remember

Reply:
![](https://i.redd.it/63hule73jam01.jpg)

Reply:
So I checked the old code, parser v1.8, and new code and found 0 references to `x-schema-private-render-type` and it being set to `true`.

So, I removed the entire extension in https://github.com/asyncapi/asyncapi-react/pull/746/commits/770197296de88118bf80de637025a8ea372716f8